app/routes.py

```diff
@@ -1,2 +1,277 @@
+    goal = Goal.query.get(goal_id)
+    if not goal:
+        return "", 404
+    
+    tasks = goal.tasks
+    json_tasks = []
+    for task in tasks:
+        serializeds_task = task.serialize() # dict
+        serializeds_task['goal_id'] = int(goal_id)
+        json_tasks.append(serializeds_task)
+        # result = {
+        #     'id': self.task_id,
+        #     'title': self.title,
+        #     'description': self.description,
+        #     'is_complete': self.completed_at != None
+        # } 
+
+    return {
+        "id": goal.goal_id,
+        "title": goal.title,
+        "tasks": json_tasks
+    }, 200
+            
+
+def order_by_title(task_response):
+    return task_response["title"]
+
+task_list_bp = Blueprint("task_list", __name__, url_prefix='/tasks')
+@task_list_bp.route('', methods=['GET', 'POST'])
+def handle_tasks():
+    if request.method == 'GET':
+        #return full list of tasks
+        tasks = Task.query.all()
+        task_response = []
+        for task in tasks:
+            task_response.append({
+                'id': task.task_id,
+                'title': task.title,
+                'description': task.description,
+                'is_complete': task.completed_at != None,
+                'goal_id': task.goal_id 
+            })
+        # allows access to keys - sort
+        arg = request.args
+        if "sort" in request.args:
+            if arg['sort'] == "asc":
+                task_response = sorted(task_response, key = order_by_title)
+            elif arg['sort'] == "desc":
+                task_response = sorted(task_response, key = order_by_title, reverse = True)
+
+        return jsonify(task_response)
+    elif request.method == 'POST':
+        request_body = request.get_json()
+        if "title" not in request_body:
+            return {
+                "details": "Invalid data"
+            }, 400
+        elif "description" not in request_body:
+            return {
+                "details": "Invalid data"
+            }, 400
+        elif "completed_at" not in request_body:
+            return {
+                "details": "Invalid data"
+            }, 400
+        stringify_format = "%a, %d %b %Y %H:%M:%S %Z"
+        if request_body["completed_at"]:
+
+            new_task = Task(
+                title = request_body['title'],
+                description = request_body['description'], 
+                completed_at = datetime.strptime(request_body["completed_at"], stringify_format)
+            )
+        else:
+             new_task = Task(
+                title = request_body['title'],
+                description = request_body['description'], 
+            )
+        db.session.add(new_task)
+        db.session.commit()
+        return{
+            'task':{
+                'id': new_task.task_id,
+                'title': new_task.title,
+                'description': new_task.description,
+                'is_complete': new_task.completed_at != None
```

Comment:
Consider moving the logic to convert completed_at to a `True` or `False` value into an instance method and use the `serialize` instance method here.